### DISCUSSION TITLE
Multi Replication Best Practice

### DISCUSSION THREAD

**Discussion by ronag** [apache/couchdb#3446] [Q&A] [answered]
Haven't found any good resource on best practice when setting up replication between multiple databases at different sites for maximum availability.

What is the best way to achieve this so that if any specific instance goes down the replication is still performed between the other ones? I assume setting up a push replication in every instance to every other instance can be rather wasteful?

**Comment by nickva** [comment, marked as answer]
You could do a bidirectional ring, so each `n` server would replicate to `n+1` and `n-1` neighbors. If one server goes down the data would still be replicated throughout the remaining ones.

Another option is an overlapping ring, where each `n` server replicates to `n+1` and `n+2` neighbors. This, of course, ramps up all the way to a complete mesh if every `n` server replicates to all the other servers.

**Reply by ronag** [reply]
Do you recommend push or pull?

**Reply by nickva** [reply]
Try push first, I think I've done that in the past and seen others use it more than pull.
